onnxruntime/core/providers/cpu/tensor/upsample.cc

```diff
@@ -104,7 +114,7 @@ Status UpsampleNearest(const T* input,
       int64_t input_dim0_inx = get_nearest_pixel(original_0_idx, scales[0] < 1);
       if (input_dim0_inx > input_shape[0] - 1) input_dim0_inx = input_shape[0] - 1;
       if (input_dim0_inx < 0) input_dim0_inx = 0;
-      output[output_idx++] = use_extrapolation_value[0] ? static_cast<T>(extrapolation_value) : input[input_dim0_inx];
+      output[output_idx++] = use_extrapolation_value[0] ? rounding_cast<T>(extrapolation_value) : input[input_dim0_inx];
     }
```

Comment:
nit: as the extrapolation value never changes, the call to rounding_cast on it could just be done once outside of any loop that uses it.

Reply:
Fixed. Btw, I noticed that `UpsampleMode::CUBIC` always assumes the input/output are floating-point (`X->template Data<float>()`). Is this a bug? Or is cubic upsampling somehow only supported for fp-types?

Reply:
Currently CUBIC only supports floating point as the call to Tensor::Data will throw if the data type doesn't match. An explicit check of the input type and nice error message would be much better.